#2019-10-27 Elvis Rodrigues <elvis@elrodrigues.com>
	    - lib/timespec.h
	    Changed to simple comparison between A and B to
	    prevent bad comparison due to overflow.

--- lib/timespec.h~	2005-09-21 23:05:39.000000000 -0700
+++ lib/timespec.h	2019-10-27 00:51:54.456444000 -0700
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int secA = a.tv_sec;
+  int secB = b.tv_sec;
+  return (secA > secB) ? 1 : ((secA == secB) ? (a.tv_nsec - b.tv_nsec) : -1);
 }
 
 # if ! HAVE_DECL_NANOSLEEP
